src/dev/flang/fuir/GeneratingFUIR.java

```diff
@@ -0,0 +1,3431 @@
+       codeAt(s) == ExprKind.Call   ||
+       codeAt(s) == ExprKind.Assign    ,
+       tclazz >= CLAZZ_BASE &&
+       tclazz < CLAZZ_BASE  + _clazzes.size());
+
+    int innerClazz;
+    if (accessIsDynamic(s))
+      {
+        var inner = accessedClazz(s, id2clazz(tclazz));
+        innerClazz = inner == null ? NO_CLAZZ : inner._id;
+        if (inner != null)
+          {
+            addToAccessedClazzes(s, tclazz, innerClazz);
+          }
+        /*
+        innerClazz = NO_CLAZZ;
+        var ccs = accessedClazzes(s);
+        //System.out.println("tclazz "+clazzAsString(tclazz)+" count "+ccs.length);
+        if (CHECKS) check
+          (ccs.length % 2 == 0);
+        for (var i = 0; i < ccs.length; i += 2)
+          {
+            var tt = ccs[i+0];
+            var cc = ccs[i+1];
+            if (tt == tclazz)
+              {
+                innerClazz = cc;
+              }
+            //  System.out.println("tclazz "+clazzAsString(tclazz)+" vs tt "+clazzAsString(tt));
+          }
+        if (CHECKS) check
+          (innerClazz != NO_CLAZZ);
+        */
+      }
+    else
+      {
+        innerClazz = accessedClazz(s);
+        if (tclazz != clazzOuterClazz(innerClazz))
+          System.out.println("static : tclazz "+
+                             clazzAsString(tclazz)+" vs "+
+                             clazzAsString(clazzOuterClazz(innerClazz))+
+                             clazzIsRef(tclazz)+" vs "+
+                             clazzIsRef(clazzOuterClazz(innerClazz))+
+                             " "+(tclazz-CLAZZ_BASE)+" vs "+(clazzOuterClazz(innerClazz)-CLAZZ_BASE)+
+                             " compare: "+id2clazz(tclazz).compareTo(id2clazz(clazzOuterClazz(innerClazz)))+" "+
+                             " inner "+clazzAsString(innerClazz)+" "+clazzKind(innerClazz)+" from "+id2clazz(innerClazz).feature().pos().show());
+        if (CHECKS) check
+          (tclazz == clazzOuterClazz(innerClazz));
+      }
+    if (innerClazz != NO_CLAZZ)
+      {
+        innerClazz = switch (clazzKind(innerClazz))
+          {
+          case Routine, Intrinsic, Native, Field -> innerClazz;
+          case Abstract, Choice -> NO_CLAZZ;
+          };
+      }
+    if (innerClazz != NO_CLAZZ && codeAt(s) == ExprKind.Call)
+      {
+        doesNeedCode(innerClazz);
+      }
+
+    // System.out.println("LOOKUP for "+clazzAsString(tclazz)+" is "+clazzAsString(innerClazz)+" at "+sitePos(s).show());
+    return innerClazz;
+  }
+
+
+  /**
+   * Inform the FUIR instance that lookup for new clazzes is finished.  This
+   * means that clazzIsUnitType will be able to produce correct results since no
+   * more features will be added.
+   */
+  @Override
+  public void lookupDone()
+  {
+    if (CHECKS) check
+      (!_lookupDone);
+
+    _lookupDone = true;
+
+    // NYI: layout phase creates new clazzes, which is why we cannot iterate like this. Need to check why and remove this!
+    //
+    // for(var c : _clazzes)
```

Comment:
orphaned code?

Reply:
actually not, I would like to use `var c : _clazzes`, but currently cant. will extend the NYI-comment